### PR TITLE
Add API to set advertising scheme on disconnected

### PR DESCRIPTION
**Pull request by XuGuohui**
### Problem
Currently when device acting as BLE peripheral is disconnected from peer device, it will re-start advertising automatically. In some use cases, user may wants to stop advertising on disconnected, unless otherwise user manually calls `BLE.advertise()` to start advertising.

### Solution
Add wiring API to set advertising scheme for which after disconnected from peer deivce.

### Example App

```cpp
#include "application.h"

SYSTEM_MODE(MANUAL);

SerialLogHandler logHandler(LOG_LEVEL_ALL);

system_tick_t disconTime = 0;
bool restartAdv = false;

/* executes once at startup */
void setup() {
    // BLE.setAdvertisingScheme(BleAdvertisingScheme::RESTART_ADV_ON_DISCONNECTED);
    // BLE.setAdvertisingScheme(BleAdvertisingScheme::STOP_ADV_ON_DISCONNECTED);
    BLE.setAdvertisingScheme(BleAdvertisingScheme::STOP_ADV_ON_DISCONNECTED_ONCE);
    BLE.advertise();
    BLE.onDisconnected([&](const BlePeerDevice& peer) {
        disconTime = millis();
        restartAdv = true;
    });
}

/* executes continuously after setup() runs */
void loop() {
    if (restartAdv && (millis() - disconTime > 15000)) {
        restartAdv = false;
        if (!BLE.advertising()) {
            Log.info("Restarting advertising");
            BLE.advertise();
        }
    }
}
```

### References
https://community.particle.io/t/ble-disconnect-works-but-then-reconnects/68469

---

### Completeness

- [x] User is totes amazing for contributing!
- [x] Contributor has signed CLA ([Info here](https://github.com/spark/firmware/blob/develop/CONTRIBUTING.md))
- [x] Problem and Solution clearly stated
- [ ] Run unit/integration/application tests on device
- [ ] Added documentation
- [ ] Added to CHANGELOG.md after merging (add links to docs and issues)

### CODE REVIEW

**Comment by XuGuohui** [comment]
TODO: I'll add on-device tests after we approve the wiring API prototypes.

**Review by avtolstoy — COMMENTED** [review]
@XuGuohui Can we add a test for this to ble central/peripheral suite?

File: wiring/inc/spark_wiring_ble.h [hunk 1]
```diff
@@ -1010,6 +1016,9 @@ class BleLocalDevice {
     ssize_t getScanResponseData(BleAdvertisingData* scanResponse) const;
     ssize_t getScanResponseData(BleAdvertisingData& scanResponse) const;
 
+    int setAdvertisingScheme(BleAdvertisingScheme scheme) const;
+    int getAdvertisingScheme(BleAdvertisingScheme* scheme) const;
```
Comment (avtolstoy) [inline wiring/inc/spark_wiring_ble.h:1020]: Similar to other cases let's add both pointer and reference getter perhaps?
```
int getScanParameters(BleScanParams* params) const;
int getScanParameters(BleScanParams& params) const;
```

**Review by avtolstoy — APPROVED** [review]

File: wiring/inc/spark_wiring_ble.h (hunk 1 above)
Comment (XuGuohui) [reply wiring/inc/spark_wiring_ble.h:1020]: Sounds good.

**Pull request merged**